File: Python/720. Longest Word in Dictionary.py
class Solution:
    def longestWord(self, words):
        """
        :type words: List[str]
        :rtype: str
        """
        words.sort()
        s, ans = set(), ''
        for word in words:
            if len(word) == 1 or word[:-1] in s:
                s.add(word)
                if len(word) > len(ans):
                    ans = word
        return ans

File: C++/720. Longest Word in Dictionary.cpp
class Solution {
public:
    string longestWord(vector<string>& words) {
        set<string> s;
        sort(words.begin(),words.end());
        string ans;
        for(auto word: words)
        {
            if(word.size()==1||s.find(word.substr(0,word.size()-1)) != s.end())
            {
                s.insert(word);
                if(word.size() > ans.size())
                    ans = word;
            }
        }
        return ans;
    }
};